### PR TITLE
ServicesPayment in Starlight runtime

### PR DESCRIPTION
**Pull request by fgamundi**
Adds `ServicePayment` pallet to Starlight runtime. This also enables service charging on collator assignment, free credits on registration and removal of assignment on insufficient credits.

### CODE REVIEW

**Comment by tmpolaczyk** [comment]
zombie_tanssi_relay tests are failing because parachain 1000 is not producing blocks. Maybe it doesn't have credits?

File: solo-chains/runtime/starlight/src/tests/common/mod.rs [hunk 1]
```diff
@@ -425,6 +427,42 @@ impl ExtBuilder {
...
+                        registered_para.para_id.into(),
+                        registered_para.block_production_credits,
+                        registered_para.collator_assignment_credits,
+                    )
+                        .into()
+                })
+                .collect(),
+        }
+        .assimilate_storage(&mut t)
+        .unwrap();
```
Comment (tmpolaczyk) [inline solo-chains/runtime/starlight/src/tests/common/mod.rs:464]: This block is duplicated, I believe it's identical?

File: solo-chains/runtime/starlight/README.md [hunk 2]
```diff
@@ -3,10 +3,9 @@
 Starlight is a testnet runtime with no stability guarantees.
 
 ## How to build `starlight` runtime
-`EpochDurationInBlocks` parameter is configurable via `STARLIGHT_EPOCH_DURATION` environment variable. To build wasm
-runtime blob with customized epoch duration the following command shall be executed:
+To build wasm runtime blob with customized epoch duration the following command shall be executed:
 ```bash
-STARLIGHT_EPOCH_DURATION=10 ./polkadot/scripts/build-only-wasm.sh starlight-runtime /path/to/output/directory/
+./polkadot/scripts/build-only-wasm.sh starlight-runtime /path/to/output/directory/
```
Comment (tmpolaczyk) [inline solo-chains/runtime/starlight/README.md:8]: Why this change?

**Review by tmpolaczyk — APPROVED** [review]

File: solo-chains/runtime/starlight/src/lib.rs [hunk 3]
```diff
@@ -1471,10 +1516,9 @@ impl pallet_registrar::RegistrarHooks for StarlightRegistrarHooks {
         // Remove bootnodes from pallet_data_preservers
         DataPreservers::para_deregistered(para_id);
 
-        ServicesPayment::para_deregistered(para_id);
-
         XcmCoreBuyer::para_deregistered(para_id);
          */
+        ServicesPayment::para_deregistered(para_id);
 
         Weight::default()
```
Comment (ParthDesai) [inline solo-chains/runtime/starlight/src/lib.rs:1523]: It seems we are not using this returned weight. Plus the pallet specific handlers also not returning `Weight`, Can we remove this in future PRs?
